test/availability/middleware/service.availability.spec.unit.ts

```diff
@@ -0,0 +1,57 @@
+import * as request from "supertest";
+import app from "../../../src/app";
+import { COOKIE_NAME } from "../../../src/properties";
+import { loadSession } from "../../../src/services/redis.service";
+import { loadMockSession } from "../../mock.utils";
+
+jest.mock("../../../src/session/store/redis.store", () => import("../../mocks/redis.store.mock.factory"));
+jest.mock("../../../src/services/redis.service");
+jest.mock("../../../src/services/session.service");
+const mockCacheService = loadSession as jest.Mock;
+
+beforeEach(() => {
+  loadMockSession(mockCacheService);
+});
+
+afterAll(() => {
+  process.env.SHOW_SERVICE_UNAVAILABLE_PAGE = "off";
+});
+
+const UNAVAILABLE_TEXT = "service is unavailable";
+
+describe("Availability tests", () => {
+
+  it("should show the service unavailable page", async () => {
+    process.env.SHOW_SERVICE_UNAVAILABLE_PAGE = "on";
+
+    const response = await request(app)
+      .get("/company-required");
+    expect(response.text).toContain(UNAVAILABLE_TEXT);
+  });
+
+  it("should show the service unavailable page with slash", async () => {
+    process.env.SHOW_SERVICE_UNAVAILABLE_PAGE = "on";
+
+    const response = await request(app)
+      .get("/company-required/");
+    expect(response.text).toContain(UNAVAILABLE_TEXT);
+  });
+
+  it("should show the service unavailable page for non start page", async () => {
+    process.env.SHOW_SERVICE_UNAVAILABLE_PAGE = "on";
+
+    const response = await request(app)
+      .get("/company-required/company-number");
```

Comment:
this test may only be passing due to you not setting a referrer.
The lack or referer causes a redirect to the start page that then redirects to the unavailable page.
If a user is mid journey when we turn this flag on, do we want to allow them to finish the journey or stop them right away?

Reply:
PR looks fine to me apart from this comment. I guess we'd want to stop all journeys immediately, no? If session timeout was/is 20 mins or something then you could have people completing journeys quite a bit later when others still couldn't get in.

Reply:
yes we want to stop them right away. I'll look at adding another test with a referer in to simulate a user who's already in the app.

Reply:
If we'd like to stop them right away then in the app.ts we could just do 
`app.use(checkServiceAvailability)` instead of `app.use(`${pageURLs.COMPANY_REQUIRED}`, checkServiceAvailability);`

That way the middleware will run on all url's

Reply:
code updated to check all urls and added extra test with referer